src/tracks/playabletrack/wavetrack/ui/WaveTrackShifter.cpp

```diff
@@ -25,9 +25,23 @@ class WaveTrackShifter final : public TrackShifter {
       auto pClip = [&]() {
          for (auto clip : mpTrack->GetClips())
          {
-            if (WaveTrackView::HitTest(*clip, viewInfo, params->rect, { params->xx, params->yy }))
-               return clip;
+            if (params != nullptr)
+            {
+               if (WaveTrackView::HitTest(
+                      *clip, viewInfo, params->rect,
+                      { params->xx, params->yy }))
+                  return clip;
+            }
+            else
+            {
+               // WithinPlayRegion misses first sample, which breaks moving
+               // "selected" clip. Probable WithinPlayRegion should be fixed
+               // instead?
+               if (clip->GetPlayStartTime() <= time && time < clip->GetPlayEndTime())
```

Comment:
The condition for comparing clip boundaries used earlier can be found here: https://github.com/audacity/audacity/commit/69681f84d5f2943e6a5c1361bb3f0a10dcf2de24
but I do not see much difference, since the original condition has already changed